mono/utils/mono-threads-windows.c

```diff
@@ -394,4 +397,74 @@ mono_native_thread_set_name (MonoNativeThreadId tid, const char *name)
 #endif
 }
 
+void
+mono_native_thread_set_namew (HANDLE threadHandle, const wchar_t *name)
+{
+// Bing: SetThreadDescription.
+// https://msdn.microsoft.com/en-us/library/windows/desktop/mt774976(v=vs.85).aspx
+// This mechanism has the following advantages over the older mechanism:
+// - Supported by the kernel, not debuggers.
+// - It works when debugger is not attached.
+//   - Either before debugger is attached.
+//   - Between detach and reattach.
+//   - No debugger at all.
+//   - This is arguably a memory footprint disadvantage.
+// - There is an API to get the name, instead of being buried in the debugger.
+// - ETW knows about it, so thread names appear in ETW traces.
+// - Unicode (only) support.
+// - Implemented by Mono team.
```

Comment:
Not clearly an advantage, and maybe just distill long comment down to the two links, or just one Bruce, anyone can Bing:SetThreadDescription.